src/python/WMCore/WorkQueue/Policy/Start/Block.py

```diff
@@ -222,12 +221,12 @@ def getMaskedBlocks(self, task, dbs, datasetPath):
                 lfn = fileLumi['logical_file_name']
                 runNumber = str(fileLumi['run_num'])
                 lumis = fileLumi['lumi_section_num']
-                if lumiMask.get(runNumber, False):
-                    if len(set(lumis) & set(lumiMask[runNumber])) > 0:
-                        maskedBlocks.setdefault(block, {})
-                        maskedBlocks[block].setdefault(lfn, LumiList())
-                        lumiList = LumiList(runsAndLumis = {runNumber: lumis})
-                        maskedBlocks[block][lfn] += (lumiList & taskMask)
+                fileMask = LumiList(runsAndLumis={runNumber: lumis})
+                commonMask = taskMask & fileMask
+                if commonMask:
+                    maskedBlocks.setdefault(block, {})
+                    maskedBlocks[block].setdefault(lfn, LumiList())
+                    maskedBlocks[block][lfn] += commonMask
 
```

Comment:
Or is this all in a loop ending here so we build it up in the final data structure?